pkg/kubelet/cm/deviceplugin/manager.go

```diff
@@ -155,37 +149,6 @@ func (m *ManagerImpl) genericDeviceUpdateCallback(resourceName string, added, up
 	m.writeCheckpoint()
 }
 
-func (m *ManagerImpl) removeContents(dir string) error {
-	d, err := os.Open(dir)
-	if err != nil {
-		return err
-	}
-	defer d.Close()
-	names, err := d.Readdirnames(-1)
-	if err != nil {
-		return err
-	}
-	for _, name := range names {
-		filePath := filepath.Join(dir, name)
-		if filePath == m.checkpointFile() {
-			continue
-		}
-		stat, err := os.Stat(filePath)
-		if err != nil {
-			glog.Errorf("Failed to stat file %v: %v", filePath, err)
-			continue
-		}
-		if stat.IsDir() {
-			continue
-		}
-		err = os.RemoveAll(filePath)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
 const (
 	// kubeletDevicePluginCheckpoint is the file name of device plugin checkpoint
 	kubeletDevicePluginCheckpoint = "kubelet_internal_checkpoint"
```

Comment:
Should we consider to move checkpoint file to a separate location, like /var/run/kubelet/device-plugins/?

Reply:
may be in next PR?